### PR TITLE
e2e/systemd: test with enforcing selinux

### PR DESCRIPTION
**Pull request by EmilienM**
**e2e/systemd: test with enforcing selinux**
Run the e2e/systemd_test with SElinux in enforcing mode.
It'll just enable the mode during the test, and restore previous mode at
the end of the test.
Also, the test will be skipped if SElinux isn't enabled on the host

**e2e/systemd: move userns skip to BeforeEach**
Moving the skip that check if userns is enabled to the BeforeEach
will help to skip the test quicker if not needed and also allow to add
more tests without adding the check everytime.

### CODE REVIEW

**Comment by mheon** [comment]
F28 looks in infra issues
/approve

**Comment by mheon** [comment]
We're waiting on https://github.com/containers/container-selinux/pull/59 to merge before this is ready - WIP will come off once that's good

**Comment by giuseppe** [comment]
what is the status of this PR?  @EmilienM are you still working on it?

**Comment by EmilienM** [comment]
> 
> 
> what is the status of this PR? @EmilienM are you still working on it?

no I have not been catching up on that one, mostly doing it on my free time. I still see some value having it, please give me a bit of time.

**Comment by rhatdan** [comment]
@EmilienM Is this still something you want/need or has it been fixed in the testing?

**Comment by EmilienM** [comment]
> @EmilienM Is this still something you want/need or has it been fixed in the testing?

Just rebased. I'll look at the tests this week if they fail and if I have the time otherwise close it.

**Comment by EmilienM** [comment]
@rhatdan ready for review it seems

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@EmilienM Not sure what happened here, could you rebase and repush so we can get the CI/CD system running again.

File: test/e2e/systemd_test.go [hunk 1]
```diff
@@ -47,11 +51,12 @@ WantedBy=multi-user.target
...
-	It("podman start container by systemd", func() {
-		if os.Getenv("SKIP_USERNS") != "" {
-			Skip("Skip userns tests.")
+	It("podman start container by systemd with selinux enforced", func() {
+		if !selinux.GetEnabled() {
+			Skip("SELinux not enabled")
 		}
-
+		mode := selinux.DefaultEnforceMode()
+		selinux.SetEnforceMode(1)
```
Comment (giuseppe) [inline test/e2e/systemd_test.go]: could you change it to something like:

```
mode := selinux.EnforceMode()
if mode != selinux.Enforcing {
    err := selinux.SetEnforceMode(selinux.Enforcing)
    Expect(err).To(BeNil())
    defer selinux.SetEnforceMode(mode)
}
```
then you can also drop the `selinux.SetEnforceMode(mode)` at line 92

**Comment by rhatdan** [comment]
@EmilienM Can you make the change requested by @giuseppe ?

**Comment by EmilienM** [comment]
I broke the PR by accident, let me fix that

**Comment by EmilienM** [comment]
I'm going to create a new PR

**Comment by EmilienM** [comment]
https://github.com/containers/libpod/pull/3727

**Pull request closed without merging**